src/main/resources/data/computercraft/lua/rom/programs/poster.lua

```diff
@@ -101,82 +180,87 @@ local function commitPrint(data)
-  error("Printer is busy", 0)
+local function printRow(name, value, optional)
+  term.setTextColor(highlightColor)
+  term.write(name)
+  term.setTextColor(colors.white)
+  term.write(value)
+  if optional then
+    term.setTextColor(colors.lightGray)
+    term.write(optional)
+  end
 end
 
-commitPrint(printData)
+-- Loop to print all selected posters
+local printed = 0
+local maxPages = #posters*count
+local numDigits = math.floor(math.log10(maxPages))+1
+local terminated = false
 
-term.setTextColor(colors.yellow)
-print("\nHold Ctrl+T to stop\n")
-showPrintStatus()
-
-while true do
-  local e, p1 = os.pullEventRaw()
-  local n = math.min(math.max((count - remaining) + 1, 1), count)
+for i, data in ipairs(posters) do
+  local currentPage = 0
+  if terminated then break end
+  commitPrint(data)
+  
+  term.setCursorPos(1, ypos)
+  printRow("Name: ", (data.label or "???"):sub(1, width-6))
+  
+  term.setCursorPos(1, ypos+1)
+  printRow("Status: ", ("  0%% : %"..numDigits.."d / %"..numDigits.."d"):format(printed, maxPages), (" (x%d)       "):format(count))
 
-  if e == "poster_printer_complete" or e == "poster_printer_state" then
-    remaining = p1
-    showPrintStatus()
 
-    if remaining <= 0 then
-      -- Printing complete, print the count we managed to print
-      local _, y = term.getCursorPos()
-      term.setCursorPos(1, y)
-      term.clearLine()
-      term.setTextColor(colors.green)
-      print("Printed " .. n .. " item" .. (n ~= 1 and "s" or ""))
+  local cur, max = printer.getInkLevel()
+  term.setCursorPos(1, ypos+2)
+  printRow("Ink: ", ("%3d%%   "):format(math.floor(100/max*cur)))
+  
+  -- Status and Ink level
+  while true do
+    local e, p1 = os.pullEventRaw()
+    
+    if e == "poster_printer_complete" or e == "poster_printer_state" then
+      local progress = 100-(tonumber(p1) or 0)
+      
+      if progress >= 100 then
+        printed = printed+1
+        currentPage = currentPage+1
+      end
 
+      term.setCursorPos(1, ypos+1)
+      printRow("Status: ", ("%3d%% : %"..numDigits.."d / %"..numDigits.."d "):format(progress, printed, maxPages), ("(x%d)       "):format(count))
+      if currentPage >= count then
+        break
+      end
+    elseif e == "terminate" then
+      printer.stop()
+      terminated = true
       break
     end
-  elseif e == "terminate" then    
-    -- Halt printing and print the count we managed to print
-    printer.stop()
-
-    local _, y = term.getCursorPos()
-    term.setCursorPos(1, y)
-    term.clearLine()
-    term.setTextColor(colors.red)
-    print("Printing terminated. Printed " .. n .. " item" .. 
-      (n ~= 1 and "s" or "") .. " out of " .. count)
-
-    break 
   end
 end
 
-term.setTextColor(startColor)
+-- Print results
+term.setCursorPos(1, ypos+3)
+if terminated then
+  term.setTextColor(errColor)
+  print(("Printed %d poster out of %d."):format(printed+1, maxPages))
+else
+  term.setTextColor(greenColor)
+  print(("Printed %d poster!"):format(printed))
```

Comment:
This is still incorrect. It should be `poster` if equal to 1, `posters` otherwise.